lib/internal/fs/streams.js

```diff
@@ -65,6 +68,36 @@ function _construct(callback) {
   }
 }
 
+// This generates an fs operations structure for a FileHandle
+const FileHandleOperations = (handle) => {
+  return {
+    open: (path, flags, mode, cb) => {
+      throw new ERR_METHOD_NOT_IMPLEMENTED('open()');
+    },
+    close: (fd, cb) => {
+      handle[kUnref]();
+      handle.close()
+        .then(() => cb())
+        .catch((err) => cb(err));
+    },
+    read: (fd, buf, offset, length, pos, cb) => {
+      handle.read(buf, offset, length, pos)
+        .then((r) => cb(null, r.bytesRead, r.buffer))
+        .catch((err) => cb(err, 0, buf));
+    },
+    write: (fd, buf, offset, length, pos, cb) => {
+      handle.read(buf, offset, length, pos)
+        .then((r) => cb(null, r.bytesWritten, r.buffer))
+        .catch((err) => cb(err, 0, buf));
```

Comment:
```suggestion
      PromisePrototypeThen(handle.read(buf, offset, length, pos),
                           (r) => cb(null, r.bytesWritten, r.buffer), 
                           (err) => cb(err, 0, buf));
```
Shouldn't this be `write` instead of `read`?
```suggestion
      PromisePrototypeThen(handle.write(buf, offset, length, pos),
                           (r) => cb(null, r.bytesWritten, r.buffer), 
                           (err) => cb(err, 0, buf));
```

Reply:
oops